SSA/Projects/RISCV64/PrettyEDSL.lean

```diff
@@ -0,0 +1,185 @@
+assembly-like style.
+
+[RV64_com| {
+  ^bb0(%r1 : !i64, %r2 : !i64 ):
+  %1 =  const (0) : !i64
+  %2 =  sub %r1,  %1 : !i64
+        ret %2 : !i64
+
+We use the functionalities defined in `SSA.Core.MLIRSyntax.PrettyEDSL`.
+For operands with immediates or shift amounts encoded as part of the opcode,
+we define additional rules.
+ -/
+
+open MLIR.AST
+open Lean
+
+syntax "add" : MLIR.Pretty.uniform_op
+syntax "sub" : MLIR.Pretty.uniform_op
+syntax "and" : MLIR.Pretty.uniform_op
+syntax "or" : MLIR.Pretty.uniform_op
+syntax "xor" : MLIR.Pretty.uniform_op
+syntax "sll" : MLIR.Pretty.uniform_op
+syntax "sra" : MLIR.Pretty.uniform_op
+syntax "mul" : MLIR.Pretty.uniform_op
+syntax "div" : MLIR.Pretty.uniform_op
+syntax "divu" : MLIR.Pretty.uniform_op
+syntax "remu" : MLIR.Pretty.uniform_op
+syntax "rem" : MLIR.Pretty.uniform_op
+syntax "li" : MLIR.Pretty.uniform_op
+syntax "ret" : MLIR.Pretty.uniform_op
+syntax "sext.b" : MLIR.Pretty.uniform_op
+syntax "sext.h" : MLIR.Pretty.uniform_op
+syntax "zext.h" : MLIR.Pretty.uniform_op
+syntax "ror" : MLIR.Pretty.uniform_op
+syntax "rol" : MLIR.Pretty.uniform_op
+syntax "addw" : MLIR.Pretty.uniform_op
+syntax "subw" : MLIR.Pretty.uniform_op
+syntax "sllw" : MLIR.Pretty.uniform_op
+syntax "srlw" : MLIR.Pretty.uniform_op
+syntax "sraw" : MLIR.Pretty.uniform_op
+syntax "srl" : MLIR.Pretty.uniform_op
+syntax "slt" : MLIR.Pretty.uniform_op
+syntax "sltu" : MLIR.Pretty.uniform_op
+syntax "czero.eqz" : MLIR.Pretty.uniform_op
+syntax "czero.nez" : MLIR.Pretty.uniform_op
+syntax "bclr" : MLIR.Pretty.uniform_op
+syntax "bext" : MLIR.Pretty.uniform_op
+syntax "binv" : MLIR.Pretty.uniform_op
+syntax "bset" : MLIR.Pretty.uniform_op
+syntax "rolw" : MLIR.Pretty.uniform_op
+syntax "rorw" : MLIR.Pretty.uniform_op
+syntax "mulu" : MLIR.Pretty.uniform_op
+syntax "mulh" : MLIR.Pretty.uniform_op
+syntax "mulhu" : MLIR.Pretty.uniform_op
+syntax "mulhsu" : MLIR.Pretty.uniform_op
+syntax "mulw" : MLIR.Pretty.uniform_op
+syntax "divw" : MLIR.Pretty.uniform_op
+syntax "divwu" : MLIR.Pretty.uniform_op
+syntax "remw" : MLIR.Pretty.uniform_op
+syntax "remwu" : MLIR.Pretty.uniform_op
+syntax "add.uw" : MLIR.Pretty.uniform_op
+syntax "sh1add.uw" : MLIR.Pretty.uniform_op
+syntax "sh2add.uw" : MLIR.Pretty.uniform_op
+syntax "sh3add.uw" : MLIR.Pretty.uniform_op
+syntax "sh1add" : MLIR.Pretty.uniform_op
+syntax "sh2add" :  MLIR.Pretty.uniform_op
+syntax "sh3add" :  MLIR.Pretty.uniform_op
+
+private def test_simple := [RV64_com| {
+  ^bb0(%e1 : !i64, %e2 : !i64 ):
+  %1 =  add %e1, %e2 : !i64
+        ret %1 : !i64
+}]
+private def test_simple2 := [RV64_com| {
+  ^bb0(%e1 : !i64, %e2 : !i64 ):
+  %1 =  add %e1, %e2 : !i64
+  %2 =  sub %1, %1 : !i64
+        ret %2  : !i64
+}]
+#check test_simple
+
+/-
+Bellow we implement the case, where an operation has `one` attirbute value.
+e.g constant or single register operations.
+WIP: extend pretty print for negative numerals/immediates. Somehow always threw an error, but is wip.
```

Comment:
```suggestion
Below we implement the case, where an operation has `one` attribute value,
e.g., constant or single register operations.
WIP: extend pretty print for negative numerals/immediates. Somehow they always
throw an error, but is wip.
```

Reply:
@salinhkuhn again, here Tobias gives a very straightforward suggestion, so you can just accept the suggestion and resolve the conversation (note also the "Commit suggestion" button, which will apply the suggestion and resolve the conversion in one go, or "Add suggestion to batch" if you want to do multiple suggestions within a single commit)